### PR TITLE
feat: bump katana

### LINKED ISSUE #998: dev: bump katana, starknet_py and adopt rpc spec 0.6 in all python scripts
```[tasklist]
### Tasks
- [x] bump katana to 0.6 alpha. 1 or 2
- [x] bump starknet py to 0.20
- [ ] adapt scripts to adopt rpc spec 0.6
```

### PR DESCRIPTION
**Pull request by Eikix**
Time spent on this PR:

## Pull request type

Please check the type of change your PR introduces:

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [x] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

## What is the current behavior?

Resolves #998 

## What is the new behavior?

- remove wait_for_transaction fixture and function, use client.wait_for_tx instead
- bump starknet py and katana
- don't bump tx type to v3 because it just doesn't work (signature invalid error without any change to the payload)

- - -
This change is [Reviewable](https://reviewable.io/reviews/kkrt-labs/kakarot/999)

### CODE REVIEW

File: scripts/utils/starknet.py [hunk 1]
```diff
@@ -100,13 +98,14 @@ async def get_starknet_account(
...
-                == "Client failed with code 40: Requested entry point was not found."
-                or err.message
-                == "Client failed with code 21: Invalid message selector."
-                or "StarknetErrorCode.ENTRY_POINT_NOT_FOUND_IN_CONTRACT" in err.message
+                "Client failed with code 40: Contract error." in message
+                or "Client failed with code 40: Requested entry point was not found."
+                in message
+                or "Client failed with code 21: Invalid message selector." in message
+                or "StarknetErrorCode.ENTRY_POINT_NOT_FOUND_IN_CONTRACT" in message
+                or ("code 40" and "not found in contract") in message
```
Comment (ClementWalter) [inline scripts/utils/starknet.py:108]: `("code 40" and "not found in contract")` means only `"not found in contract"`

not sure about what you want

File: scripts/utils/starknet.py [hunk 2]
```diff
@@ -177,12 +176,15 @@ async def fund_address(
             raise ValueError(
                 f"Cannot send {amount / 1e18} ETH from default account with current balance {balance / 1e18} ETH"
             )
-        prepared = eth_contract.functions["transfer"].prepare(
+        prepared = eth_contract.functions["transfer"].prepare_invoke_v1(
             address, int_to_uint256(amount)
         )
-        tx = await prepared.invoke(max_fee=_max_fee)
+        tx = await prepared.invoke(
+            max_fee=_max_fee,
```
Comment (ClementWalter) [inline scripts/utils/starknet.py:183]: remove trailing comma

**Review by ClementWalter — CHANGES_REQUESTED** [review]
You should either remove in the scripts.constants the max retry and check interval or use them in the RPC.wait_for_tx

alternatively, you can create a wait_for_tx in the utils (as it was the case), but just being 

```
functools.partial(RPC_CLIENT.wait_for_tx, retries=NETWORK["retries"], ...
```

File: scripts/utils/starknet.py (hunk 1 above)
Comment (Eikix) [reply scripts/utils/starknet.py:108]: ahh my bad, I wanted 
```python
("code 40" in message or "not found in contract" in message)
```

**Comment by Eikix** [comment]
Done!

**Review by ClementWalter — APPROVED** [review]

**Pull request merged**